flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/data/util/DataFormatConverters.java

```diff
@@ -215,6 +215,8 @@ public static DataFormatConverter getConverterForDataType(DataType originDataTyp
                     return new LongTimestampDataConverter(precisionOfLZTS);
                 } else if (clazz == TimestampData.class) {
                     return new TimestampDataConverter(precisionOfLZTS);
+                } else if (clazz == Timestamp.class) {
+                    return new TimestampConverter(precisionOfLZTS);
```

Comment:
This conversion will lost 8 hours if in Shanghai time zone. We may need a new converter for LTZ -> java.sql.Timestamp.